simd/unit_tests/TestSIMD.cpp

```diff
@@ -277,16 +277,41 @@ class divides {
   }
 };
 
-template <class Abi>
-inline void host_check_math_ops() {
-  std::size_t constexpr n     = 11;
-  double const first_args[n]  = {1, 2, -1, 10, 0, 1, -2, 10, 0, 1, -2};
-  double const second_args[n] = {1, 2, 1, 1, 1, -3, -2, 1, 13, -3, -2};
+template <class Abi, size_t n, typename DataType>
+inline void host_check_all_math_ops(DataType const& first_args,
+                                    DataType const& second_args) {
   host_check_binary_op_all_loaders<Abi>(plus(), n, first_args, second_args);
   host_check_binary_op_all_loaders<Abi>(minus(), n, first_args, second_args);
   host_check_binary_op_all_loaders<Abi>(multiplies(), n, first_args,
                                         second_args);
-  host_check_binary_op_all_loaders<Abi>(divides(), n, first_args, second_args);
+
+  if constexpr (std::is_same_v<DataType, double>)
```

Comment:
Why is division test not available of the integral types?

Reply:
That `if` was placed because the division operation is currently only implemented for `double` type in kokkos simd.
And I am guessing that may be because the division instruction is only available for float and double types in intel intrinsics. Knowing all incoming precision types, we could emulate it if needed, but I concluded that should be an incremental work and not be part of this PR.

Reply:
Added a `TODO` there for now. Because divisions are only provided for double and float types in both intel and arm intrinsics, I think there'll need to be a decision on what an integer division should produce.